### PR TITLE
Add a mention of usernames protection to `Community/Profile badge`

### PR DESCRIPTION
**Pull request by minusQuantumNeko**
## Self-check

- [X] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)

### CODE REVIEW

File: wiki/Community/Profile_badge/en.md [hunk 1]
```diff
@@ -39,3 +39,4 @@ A side effect of this layout scheme was that the default, non-expanded height of
 
 - The first two profile badges were awarded to ::{ flag=PL }:: [niedzwiedz1124](https://osu.ppy.sh/users/9610) and ::{ flag=PL }:: [White Wolf](https://osu.ppy.sh/users/39828) on 6 September, 2009, for [winning *Tag Tournament*](https://osu.ppy.sh/community/forums/topics/17169).<!-- TODO: The changelog entry for badges being added is also on the same day, and should be mentioned. are these available online anywhere to link to? -clayton -->
 - Badges can link to other web pages for more context, like a tournament's forum page or a wiki entry.
+- Badges also protect all of the player usernames from being taken due to inactivity or them changing their current username.
```
Comment (Walavouchey) [inline wiki/Community/Profile_badge/en.md:42]:
```suggestion
- Badges prevent others from from being able to [take the player's current or past usernames](/wiki/Help_centre/Account#take-existing-username).
```

Comment (Walavouchey) [inline wiki/Community/Profile_badge/en.md:42]: where does it say that their past usernames can't be taken?

Comment (minusQuantumNeko) [reply wiki/Community/Profile_badge/en.md:42]: @Walavouchey it doesn't explicitly say that, but from personal experience I can say that for sure. My previous username could be taken before I got the badge, but now it says "already in use"

Comment (cl8n) [reply wiki/Community/Profile_badge/en.md]: `UsernameValidation::validateUsersOfUsername()` :stuck_out_tongue: 

the actual logic goes: if anyone who has the username or used to have the username also has a profile badge, then the username is locked

**Review by cl8n — APPROVED** [review]

File: wiki/Community/Profile_badge/en.md (hunk 1 above)
Comment (Hiviexd) [reply wiki/Community/Profile_badge/en.md]: actual code can be found here for future reference:

https://github.com/ppy/osu-web/blob/70d5ef18611069339f45e83cf26aaa3de8ee11f3/app/Libraries/UsernameValidation.php#L96-L99

**Review by Hiviexd — APPROVED** [review]

**Pull request merged**